store/src/java-test/com/zimbra/cs/service/mail/GetFilterRulesTest.java

```diff
@@ -173,30 +172,41 @@ public void testBug71036_SingleRuleSimpleNestedIfwithMultiConditions02() throws
+                    <sizeTest s="1" numberComparison="over" index="1"/>
+                </filterTests>
+                <filterActions>
+                    <actionFlag flagName="priority" index="0"/>
+                </filterActions>
+            </nestedRule>
+        </filterRule>
+        </filterRules>
+        */
 
         Element rules = response.getOptionalElement(MailConstants.E_FILTER_RULES);
-        //ZimbraLog.filter.info(rules.prettyPrint());
-        //ZimbraLog.filter.info(expectedSoap);
-        Assert.assertEquals(expectedSoap, rules.prettyPrint());
-
+        Assert.assertEquals("anyof", rules.getElement("filterRule").getElement("filterTests").getAttribute("condition"));
+        Assert.assertEquals("is",rules.getElement("filterRule").getElement("filterTests").getElement("headerTest").getAttribute("stringComparison"));
+        Assert.assertEquals("Subject",rules.getElement("filterRule").getElement("filterTests").getElement("headerTest").getAttribute("header"));
+        Assert.assertEquals("important",rules.getElement("filterRule").getElement("filterTests").getElement("headerTest").getAttribute("value"));
+        Assert.assertEquals("0",rules.getElement("filterRule").getElement("filterTests").getElement("headerTest").getAttribute("index"));
+        Assert.assertEquals("allof",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getAttribute("condition"));
+        Assert.assertEquals("contains",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("headerTest").getAttribute("stringComparison"));
+        Assert.assertEquals("Subject",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("headerTest").getAttribute("header"));
+        Assert.assertEquals("confidential",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("headerTest").getAttribute("value"));
+        Assert.assertEquals("0",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("headerTest").getAttribute("index"));
+        Assert.assertEquals("1",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("sizeTest").getAttribute("s"));
+        Assert.assertEquals("over",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("sizeTest").getAttribute("numberComparison"));
+        Assert.assertEquals("1",rules.getElement("filterRule").getElement("nestedRule").getElement("filterTests").getElement("sizeTest").getAttribute("index"));
+        Assert.assertEquals("priority",rules.getElement("filterRule").getElement("nestedRule").getElement("filterActions").getElement("actionFlag").getAttribute("flagName"));
+        Assert.assertEquals("0",rules.getElement("filterRule").getElement("nestedRule").getElement("filterActions").getElement("actionFlag").getAttribute("index"));
     }
```

Comment:
There any many failures similar to the above test case. I think we should use xmlunit for this. It is under apache license 2.0, and we have to use only for testing purpose. So, I don't think we will have any issues for legal approval. Basically, this will reduce our coding effort and will be reusable 
http://www.xmlunit.org/